framework/src/modules/chain/synchronizer/block_synchronization_mechanism.js

```diff
@@ -113,20 +115,47 @@ class BlockSynchronizationMechanism {
 			);
 		}
 
-		try {
-			await deleteBlocksAfterHeight(
-				this.processorModule,
-				this.blocks,
-				lastCommonBlock.height,
-			);
-		} catch (e) {
-			this.logger.debug(
-				`Failure deleting blocks after height ${
-					lastCommonBlock.height
-				} while reverting chain to last common block`,
-			);
-			throw e;
+		await deleteBlocksAfterHeightAndBackup(
+			this.processorModule,
+			this.blocks,
+			lastCommonBlock.height,
+		);
+	}
+
+	/**
+	 * Returns a list of block heights corresponding to the first block of a defined number
+	 * of rounds (listSizeLimit)
+	 *
+	 * @param listSizeLimit
+	 * @param _currentRound
```

Comment:
Multiline comment whoa!

- Can you add description for `listSizeLimit`
- I think that we don't append variables with `_` as the function is already privately defined?

Reply:
I think it makes sense to append `_` to a variable name when it needs to be cloned inside the function scope. It's done like this in the Storage layer.

Reply:
Done :)